Java/validation/src/main/java/com/lantanagroup/link/validation/services/PreQualService.java

```diff
@@ -0,0 +1,206 @@
+package com.lantanagroup.link.validation.services;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.lantanagroup.link.validation.entities.Category;
+import com.lantanagroup.link.validation.entities.PreQualSummary;
+import com.lantanagroup.link.validation.entities.Result;
+import lombok.Getter;
+import lombok.Setter;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.text.StringEscapeUtils;
+import org.apache.commons.text.StringSubstitutor;
+import org.hl7.fhir.r4.model.OperationOutcome;
+import org.springframework.context.annotation.Scope;
+import org.springframework.context.annotation.ScopedProxyMode;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+@Service
+@Scope(value = "prototype", proxyMode = ScopedProxyMode.TARGET_CLASS)
+public class PreQualService {
+
+    private Map<String, Category> categoriesById = new HashMap<>();
+    private Map<Category, Integer> countsByCategory = new HashMap<>();
+    private Map<Result, Integer> countsByIssue = new HashMap<>();
```

Comment:
Not critical, but recommend converting these fields to local variables in `generateSimplePreQualReport` (and passing them to other methods as required).  At that point, we'd no longer need prototype-scoping on the class (i.e., can remove the `@Scope` annotation), which makes more sense IMO.